src/Credentials/GCECredentials.php

```diff
@@ -395,6 +412,28 @@ private static function detectResidencyLinux(string $productNameFile): bool
         return false;
     }
 
+    private static function detectResidencyWindows(string $registryProductKey): bool
+    {
+        if (!class_exists(COM::class)) {
+            // the COM extension must be installed and enabled to detect Windows residency
+            // see https://www.php.net/manual/en/book.com.php
+            return false;
+        }
+
+        $shell = new COM('WScript.Shell');
+        $productName = null;
+
+        try {
+            $productName = $shell->regRead($registryProductKey);
+        } catch(com_exception $e) {
+            // This means that we tried to read a key that doesn't exist on the registry
+            // which might mean that it is a windows instance that is not on GCE
+            return false;
+        }
+        
+        return explode(' ', $productName)[0] === self::WINDOWS_PRODUCT_NAME;
```

Comment:
nit: I think we should be consistent with how we do this in `detectResidencyLinux`